src/openrct2/object/RideObject.cpp

```diff
@@ -229,16 +231,32 @@ void RideObject::Load()
             uint32_t imageIndex = baseImageId;
             carEntry->base_image_id = baseImageId;
 
-            for (uint8_t spriteGroup = 0; spriteGroup < EnumValue(SpriteGroupType::Count); spriteGroup++)
+            switch (carEntry->PaintStyle)
             {
-                if (carEntry->SpriteGroups[spriteGroup].Enabled())
-                {
-                    carEntry->SpriteGroups[spriteGroup].imageId = imageIndex;
-                    const auto spriteCount = carEntry->base_num_frames
-                        * carEntry->NumRotationSprites(static_cast<SpriteGroupType>(spriteGroup))
-                        * SpriteGroupMultiplier[spriteGroup];
-                    imageIndex += spriteCount;
-                }
+                case VEHICLE_VISUAL_RIVER_RAPIDS:
+                case VEHICLE_VISUAL_VIRGINIA_REEL:
+                    // Paint code for these rides do not use sprite groups but #17909 requires them. Dummy sprite groups are
+                    // added.
+                    carEntry->SpriteGroups[EnumValue(SpriteGroupType::SlopeFlat)] = { baseImageId, SpritePrecision::Sprites1 };
+                    carEntry->SpriteGroups[EnumValue(SpriteGroupType::Slopes12)] = {
+                        baseImageId + RiverRapidsNumSpinningSprites, SpritePrecision::Sprites4
+                    };
+                    carEntry->SpriteGroups[EnumValue(SpriteGroupType::Slopes25)] = {
+                        baseImageId + RiverRapidsNumSpinningSprites * (NumOrthogonalDirections + 1), SpritePrecision::Sprites4
```

Comment:
It now kinda looks like there are 5 directions, while 1 of the 5 is for its base offset. Though longer, perhaps this makes it easier to understand:

```suggestion
                        baseImageId + RiverRapidsNumSpinningSprites + RiverRapidsNumSpinningSprites * NumOrthogonalDirections, SpritePrecision::Sprites4
```

Also pinging @Gymnasiast

(I'm heading off to bed now, feel free to merge once this is addressed)